tests/test_fields.py

```diff
@@ -0,0 +1,98 @@
+import decimal
+import typing
+
+import attr
+import marshmallow
+import pytest
+
+import desert._fields
+
+
+# TODO: test that field constructor doesn't tromple Field parameters
+
+
+@attr.s(auto_attribs=True)
+class ExampleData:
+    object: typing.Any
+    tag: str
+    field: typing.Callable[[], marshmallow.fields.Field]
+
+
+example_data_list = [
+    ExampleData(object=3.7, tag="float_tag", field=marshmallow.fields.Float),
+    ExampleData(object="29", tag="str_tag", field=marshmallow.fields.String),
+    ExampleData(
+        object=decimal.Decimal("4.2"),
+        tag="decimal_tag",
+        field=marshmallow.fields.Decimal,
+    ),
+    ExampleData(
+        object=[1, 2, 3],
+        tag="integer_list_tag",
+        field=lambda: marshmallow.fields.List(marshmallow.fields.Integer()),
+    ),
+    ExampleData(
+        object=['abc', '2', 'mno'],
+        tag="string_list_tag",
+        field=lambda: marshmallow.fields.List(marshmallow.fields.String()),
+    ),
+]
+
+
+@pytest.fixture(
+    name="example_data",
+    params=example_data_list,
+    ids=[str(example) for example in example_data_list],
+)
+def _example_data(request):
+    return request.param
+
+
+@pytest.fixture(name="registry", scope="session")
```

Comment:
Any particular reason the fixtures are session-scoped? I can imagine accidentally mutating one in some test and having a tough time debugging.

Reply:
Premature optimization of course!

Reply:
Looks like I dropped the scoping.